src/course-unit/xblock-container-iframe/utils.ts

```diff
@@ -31,3 +31,12 @@ export const formatAccessManagedXBlockData = (
  * @returns {string} - The generated iframe URL.
  */
 export const getIframeUrl = (blockId: string): string => `${getConfig().STUDIO_BASE_URL}/container_embed/${blockId}`;
+
+/**
+ * Generates the legacy edit modal URL for the given block ID.
+ *
+ * @param {string | null} blockId - The unique identifier of the block.
+ *
+ * @returns {string} - The generated URL for editing the XBlock in the legacy modal.
+ */
+export const getLegacyEditModalUrl = (blockId: string | null): string => `${getConfig().STUDIO_BASE_URL}/xblock/${blockId}/action/edit`;
```

Comment:
[question] Is there a case where `blockId` should be undefined and won't that result in an error?

Reply:
The value of `blockId` is `configureXBlockId` and it is in a hook that has a default value of `null`. Added a check